File: Backend/src/main/java/com/Ecommerce/Backend/entity/Order.java
package com.Ecommerce.Backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

import java.util.List;

@Entity
@Table(name = "orders")
@Getter
@Setter
public class Order {
    @Getter
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Getter
    private LocalDateTime orderDate;

    @Getter
    private Double total;

    @Getter
    private Integer totalItems;

    @Getter
    @Enumerated(EnumType.STRING)
    private Status status;

    @Getter
    @ManyToOne
    @JoinColumn(name="user_id", nullable = false)
    @JsonIgnoreProperties({"password", "cart", "orders", "phone", "address", "city", "zipcode"})
    private User user;

    @Getter
    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    private List<OrderItem> items;
}

File: src/types/Order.ts
export interface Order {
  id: number;
  orderDate: string;
  total: number;
  status: string;
  totalItems: number;
  user: {
    id: number;
    name: string;
    email: string;
  };
  items: {
    id: number;
    quantity: number;
    price: number;
    product: {
      id: number;
      name: string;
      imageUrl: string;
    };
  }[];
}